### PR TITLE
GetContract should also return IBAN

### PR DESCRIPTION
**Pull request by jojowork**

### CODE REVIEW

File: Civi/Contract/ActionProvider/Action/GetContract.php [hunk 1]
```diff
@@ -79,6 +79,7 @@ public function getOutputSpecification() {
       new Specification('currency',        'Date', E::ts('Currency'), false, null, null, null, false),
       new Specification('frequency_unit',        'Date', E::ts('Frequency Unit'), false, null, null, null, false),
       new Specification('frequency_interval',        'Date', E::ts('Frequency Interval'), false, null, null, null, false),
+      new Specification('iban',        'Date', E::ts('IBAN'), false, null, null, null, false),
```
Comment (bjendres) [inline Civi/Contract/ActionProvider/Action/GetContract.php:82]: this should be 'String', I suppose

File: Civi/Contract/ActionProvider/Action/GetContract.php [hunk 2]
```diff
@@ -166,6 +167,8 @@ function str_starts_with($str, $start) {
       $output->setParameter('frequency_unit', $recurring_contribution['frequency_unit']);
       $output->setParameter('frequency_interval', $recurring_contribution['frequency_interval']);
 
+      $bankaccount_reference = \civicrm_api3('BankingAccountReference', 'getSingle', ['ba_id' => $contract[U::getCustomFieldId('membership_payment.from_ba')], 'reference_type_id' => U::lookupOptionValue('civicrm_banking.reference_types', 'iban', 'id'), 'return' => ["reference"]]);
```
Comment (bjendres) [inline Civi/Contract/ActionProvider/Action/GetContract.php:170]: I'm not sure that you can just assume that this is definitely an IBAN, even though we mostly use it that way. I would either make sure that it *is* an IBAN, or refer to it as a bank account reference.

**Review by bjendres — CHANGES_REQUESTED** [review]
Looks good to me, but I have  _run_ the code, but I trust you have.

Also, see inline comments

**Comment by bjendres** [comment]
Let me know if you have questions

File: Civi/Contract/ActionProvider/Action/GetContract.php (hunk 2 above)
Comment (jojowork) [reply Civi/Contract/ActionProvider/Action/GetContract.php:170]: I tried to make sure it is an iban by setting reference_type_id to the option-value of IBAN.
Is this not sufficient? What else could I do?

File: Civi/Contract/ActionProvider/Action/GetContract.php (hunk 1 above)
Comment (jojowork) [reply Civi/Contract/ActionProvider/Action/GetContract.php]: I changed it to String

File: Civi/Contract/ActionProvider/Action/GetContract.php (hunk 2 above)
Comment (jojowork) [reply Civi/Contract/ActionProvider/Action/GetContract.php:171]: But another problem might be, that getSingle returns an error message if there is no or there are several BankingAccountReferences available for the type IBAN.

Comment (jojowork) [reply Civi/Contract/ActionProvider/Action/GetContract.php]: ok. I changed it to from getSingle to get-request

**Comment by bjendres** [comment]
Looks good, but please try if
* it works (you probably have)
* you can set an empty value to this field (``''``)

**Review by bjendres — APPROVED** [review]

**Pull request merged**